src/main/java/no/unit/nva/cristin/projects/ProjectsWrapper.java

```diff
@@ -112,17 +124,27 @@ public void setHits(List<NvaProject> hits) {
     }
 
     /**
-     * Assigns value to some field values using supplied query parameters.
+     * Assigns value to some of the field values using supplied headers and query parameters.
      *
-     * @param queryParams the query params
-     * @return ProjectsWrapper object with some of the field values set using query parameters
+     * @param headers     the headers from response
+     * @param queryParams the query params from request
+     * @return ProjectsWrapper object with some of the field values set using the supplied parameters
      */
-    public ProjectsWrapper usingQueryParams(Map<String, String> queryParams) {
+    public ProjectsWrapper usingHeadersAndQueryParams(HttpHeaders headers, Map<String, String> queryParams) {
+        this.size = getSizeHeader(headers);
         this.id = idUriFromParams(queryParams);
         this.firstRecord = indexOfFirstEntryInPageCalculatedFromParams(queryParams);
+        setFirstRecordToZeroIfExceedsMaxSizeOfResultSet();
```

Comment:
Probably simplify as:
```
        int firstRecord = indexOfFirstEntryInPageCalculatedFromParams(queryParams);
        this.firstRecord = (this.size > firstRecord) ? firstRecord : 0;
```

Reply:
I made a method with a descriptive name to avoid commenting why we do it. Should i drop the method and just use the logic directly as you imply or keep the method and change the logic in that method?

Reply:
```
    public ProjectsWrapper usingHeadersAndQueryParams(HttpHeaders headers, Map<String, String> queryParams) {
        this.size = getSizeHeader(headers);
        this.id = idUriFromParams(queryParams);
        int firstRecord = indexOfFirstEntryInPageCalculatedFromParams(queryParams);
        this.firstRecord = pageIsEmpty(firstRecord) ? firstRecord : 0;

        return this;
    }

    private boolean pageIsEmpty(int firstRecord) {
        return this.size > firstRecord;
    }
    
```

Reply:
I think this is a BadRequest. We set a parameter "size" to be less than the "firstRecord" and we get an OK and a possibly unexpected result.  If I say bring me the 12th item from a collection of 10 items the response should not be "here is the first one", but  "your request is invalid"